### PR TITLE
Add an oci image to act as a cache builder for a multistage building strategy

### PR DESCRIPTION
**Pull request by adrianriobo**
This PR will add the Containerfile require to build the cache builder which will be used as part of the multistage building strategy.

### CODE REVIEW

File: .github/workflows/build-builder.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: build-builder
+
+on:
+  push:
+    branches:
+    - main
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
```
Comment (benoitf) [inline .github/workflows/build-builder.yaml:10]: you should specify the version and not latest as it may shift and provides unexpected behaviour

Comment (adrianriobo) [reply .github/workflows/build-builder.yaml]: :+1:

Comment (adrianriobo) [reply .github/workflows/build-builder.yaml]: done

File: .github/workflows/build-builder.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: build-builder
```
Comment (benoitf) [inline .github/workflows/build-builder.yaml:1]: missing copyright headers

**Review by benoitf — APPROVED** [review]

File: .github/workflows/build-builder.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+on:
+  push:
+    branches:
+    - main
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-22.04
```
Comment (benoitf) [inline .github/workflows/build-builder.yaml:10]: should be 24.04  ?

Comment (adrianriobo) [reply .github/workflows/build-builder.yaml:10]: oh interesting I trust on PD repo so I pick the version from there :joy: let me check and update from official gh doc

File: .github/workflows/build-builder.yaml (hunk 2 above)
Comment (adrianriobo) [reply .github/workflows/build-builder.yaml:1]: :+1:  btw should I update the dates to 2024?

File: .github/workflows/build-builder.yaml (hunk 3 above)
Comment (benoitf) [reply .github/workflows/build-builder.yaml:10]: I think we're in the process of updating it

File: .github/workflows/build-builder.yaml (hunk 2 above)
Comment (adrianriobo) [reply .github/workflows/build-builder.yaml:18]: done

File: .github/workflows/build-builder.yaml (hunk 3 above)
Comment (adrianriobo) [reply .github/workflows/build-builder.yaml]: done

**Review by benoitf — APPROVED** [review]

**Pull request merged**